src/main/java/org/jitsi/videobridge/Constraints.java

```diff
@@ -0,0 +1,26 @@
+package org.jitsi.videobridge;
+
+public class Constraints
```

Comment:
Do we need a separate class for this?  Could endpoints just have 2 instances of `EndpointConstraints`? One for global and one for their own?

Reply:
We could even have a single instance of the 'global constraints' per conference and give the endpoints a read-only form of it?  That would mean instead of `Constraints` and `EndpointConstraints`, we have `EndpointConstraints` and `ReadOnlyEndpointConstraints`.  An `Endpoint` would have an instance of `EndpointConstraints` for its own, specific constraints and be given an instance of `ReadOnlyEndpointConstraints` which was the shared, global endpoint constraints for that conference?

Reply:
> Do we need a separate class for this? Could endpoints just have 2 instances of EndpointConstraints? One for global and one for their own?

You're right, this feels redundant.

> We could even have a single instance of the 'global constraints' per conference and give the endpoints a read-only form of it?

'Global constraints' are per receiver tho, or am I reading this wrong?

Reply:
Maybe I misunderstood "Global constraints", I assumed they were constraints set on, say, a conference level (as opposed to an Endpoint level) that were applied to all Endpoints?

Reply:
@bbaldino ah, I see now where's the confusion. Admittedly I chose a very misleading name.

Imagine that this `bitrateController.globalConstraints` effectively replaces the `bitrateController.maxRxFrameHeightPx ` constraint. So, it's not conference level, it's endpoint (receiver) level but for all the endpoints (senders) that the receiver receives.

We should probably rename this to something else..

Maybe a little context here would help, so for the time being I'm trying to refactor what's already there, and then add the new stuff we had been discussing.

Reply:
Ah, ok, I think I see now.  I had the scopes wrong.  The name isn't too bad, actually, maybe we just document it somewhere.  Sorry for being slow on this, I left some more comments.